src/EventStream/ServiceExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Extensions;
+
+namespace EventStream
+{
+    public static class ServiceExtensions
+    {
+        public static IServiceCollection AddInMemoryEventStream(this IServiceCollection services)
+        {
+            services.TryAddSingleton<InMemoryEventStream, InMemoryEventStream>();
+            services.TryAddTransient(typeof(EventStream), isp => isp.GetRequiredService<InMemoryEventStream>());
```

Comment:
just because I don't like acronyms, I'd go with `provider => provider....` instead of `isp`.